crdsonnet/renderEngines/ast.libsonnet

```diff
@@ -25,23 +26,36 @@ local xtd = import 'github.com/jsonnet-libs/xtd/main.libsonnet';
   },
 
   nestInParents(name, parents, field)::
-    j.object.members([
+    local startParents =
+      if jutils.isObject(field)
+      then parents[:std.length(parents) - 1]
+      else parents;
+    local startField =
+      if jutils.isField(field) || jutils.isFunction(field)
+      then field
+      else if jutils.isObject(field)
```

Comment:
I'm being forgiving in case the input is an object type, the assumption is that when it is an object and not a field, then the object is supposed to be added to the innermost parent field.